certora/WormholeRouter.spec

```diff
@@ -0,0 +1,251 @@
+// WormholeRouter.spec
+
+using WormholeRouter as router
+using DaiMock as dai
+using WormholeJoinMock as join
+
+methods {
+    domainAt(uint256) returns (bytes32) envfree
+    domains(address) returns (bytes32) envfree
+    gateways(bytes32) returns (address) envfree
+    hasDomain(bytes32) returns (bool) envfree
+    numDomains() returns (uint256) envfree
+    requestMint(join.WormholeGUID, uint256, uint256) returns (uint256) => DISPATCHER(true)
+    settle(bytes32, uint256) => DISPATCHER(true)
+    wards(address) returns (uint256) envfree
+    dai.allowance(address, address) returns (uint256) envfree
+    dai.balanceOf(address) returns (uint256) envfree
+}
+
+definition RAY() returns uint256 = 10^27;
+
+definition min_int256() returns mathint = -1 * 2^255;
+definition max_int256() returns mathint = 2^255 - 1;
+
+ghost indexesGhost(bytes32) returns uint256 {
+    init_state axiom forall bytes32 x. indexesGhost(x) == 0;
+}
+
+hook Sload uint256 v currentContract.allDomains._inner._indexes[KEY bytes32 domain] STORAGE {
+    require indexesGhost(domain) == v;
+}
+
+hook Sstore currentContract.allDomains._inner._indexes[KEY bytes32 a] uint256 n (uint256 o) STORAGE {
```

Comment:
nit: using `a` here and `domain` above for the key variable

Reply:
Not touching this one as I think you were doing changes for the invariant in your PR.
Feel free to do the variable renaming there.